### PR TITLE
IGNITE-13169 Remove Ignite bean name requirement for Spring Data Repository

### PR DESCRIPTION
**Pull request by SammyVimes**

### CODE REVIEW

File: modules/spring-data-2.0-ext/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteProxyFactory.java [hunk 1]
```diff
@@ -137,4 +139,29 @@ private IgniteProxy createIgniteProxy(Class<?> repoInterface) {
...
+    /**
+     * Helper interface that wraps getBean method.
+     */
+    @FunctionalInterface
+    private interface BeanFinder {
+        /**
+         * Get bean.
+         * @return Bean or null if {@link BeansException} was thrown.
+         */
+        default Object getBean() {
```
Comment (tkalkirill) [inline modules/spring-data-2.0-ext/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteProxyFactory.java:152]: @nullable forget

File: modules/spring-data-2.0-ext/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteProxyFactory.java [hunk 2]
```diff
@@ -137,4 +139,29 @@ private IgniteProxy createIgniteProxy(Class<?> repoInterface) {
     private String evaluateExpression(String spelExpression) {
         return (String)expressionResolver.evaluate(spelExpression, beanExpressionCtx);
     }
+
+    /**
+     * Helper interface that wraps getBean method.
```
Comment (tkalkirill) [inline modules/spring-data-2.0-ext/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteProxyFactory.java:144]: Add getBean as @link

File: modules/spring-data-2.0-ext/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteProxyFactory.java [hunk 3]
```diff
@@ -137,4 +139,29 @@ private IgniteProxy createIgniteProxy(Class<?> repoInterface) {
...
+     */
+    @FunctionalInterface
+    private interface BeanFinder {
+        /**
+         * Get bean.
```
Comment (tkalkirill) [inline modules/spring-data-2.0-ext/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteProxyFactory.java:149]: Maybe: Getting a bean.
New line after description.

File: modules/spring-data-2.0-ext/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteProxyFactory.java [hunk 4]
```diff
@@ -137,4 +139,29 @@ private IgniteProxy createIgniteProxy(Class<?> repoInterface) {
...
+        default Object getBean() {
+            try {
+                return get();
+            } catch (BeansException ex) {
+                return null;
+            }
+        }
+
+        /**
+         * Get bean.
```
Comment (tkalkirill) [inline modules/spring-data-2.0-ext/src/main/java/org/apache/ignite/springdata20/repository/support/IgniteProxyFactory.java:161]: Maybe: Getting a bean.
New line after description.

File: modules/spring-data-2.0-ext/src/test/java/org/apache/ignite/springdata/IgniteSpringDataConnectionConfigurationTest.java [hunk 5]
```diff
@@ -75,6 +76,30 @@ public void testRepositoryWithClientConfiguration() {
         checkRepositoryConfiguration(ClientConfigurationApplication.class, IgniteClientConfigRepository.class);
     }
 
+    /** Tests repository configuration in case {@link Ignite} with non default is used to access the Ignite cluster. */
```
Comment (tkalkirill) [inline modules/spring-data-2.0-ext/src/test/java/org/apache/ignite/springdata/IgniteSpringDataConnectionConfigurationTest.java:79]: Why 1 line?

File: modules/spring-data-2.0-ext/src/test/java/org/apache/ignite/springdata/IgniteSpringDataConnectionConfigurationTest.java [hunk 6]
```diff
@@ -75,6 +76,30 @@ public void testRepositoryWithClientConfiguration() {
...
+    @Test
+    public void testRepositoryWithoutIgniteInstanceParameter() {
+        checkRepositoryConfiguration(DefaultIgniteBeanApplication.class, IgniteRepositoryWithoutExplicitIgnite.class);
+    }
+
+    /** Tests repository configuration in case {@link IgniteClient} with non default name is used to access the Ignite cluster. */
```
Comment (tkalkirill) [inline modules/spring-data-2.0-ext/src/test/java/org/apache/ignite/springdata/IgniteSpringDataConnectionConfigurationTest.java:85]: Why 1 line?

File: modules/spring-data-2.0-ext/src/test/java/org/apache/ignite/springdata/IgniteSpringDataConnectionConfigurationTest.java [hunk 7]
```diff
@@ -75,6 +76,30 @@ public void testRepositoryWithClientConfiguration() {
...
+    @Test
+    public void testRepositoryWithoutIgniteClientInstanceParameter() {
+        checkRepositoryConfiguration(DefaultIgniteClientBeanApplication.class, IgniteRepositoryWithoutExplicitIgnite.class);
+    }
+
+    /** Tests repository configuration in case {@link ClientConfiguration} with non default name is used to access the Ignite cluster. */
```
Comment (tkalkirill) [inline modules/spring-data-2.0-ext/src/test/java/org/apache/ignite/springdata/IgniteSpringDataConnectionConfigurationTest.java:91]: Why 1 line?

File: modules/spring-data-2.0-ext/src/test/java/org/apache/ignite/springdata/IgniteSpringDataConnectionConfigurationTest.java [hunk 8]
```diff
@@ -75,6 +76,30 @@ public void testRepositoryWithClientConfiguration() {
...
+    @Test
+    public void testRepositoryWithoutIgnitClientConfigurationParameter() {
+        checkRepositoryConfiguration(DefaultIgniteClientConfigurationBeanApplication.class, IgniteRepositoryWithoutExplicitIgnite.class);
+    }
+
+    /** Tests repository configuration in case {@link IgniteConfiguration} with non default name is used to access the Ignite cluster. */
```
Comment (tkalkirill) [inline modules/spring-data-2.0-ext/src/test/java/org/apache/ignite/springdata/IgniteSpringDataConnectionConfigurationTest.java:97]: Why 1 line?

**Review by sashapolo — CHANGES_REQUESTED** [review]

**Comment by sashapolo** [comment]
All comments are applicable for all three copy-pasted modules

**Pull request closed without merging**